smack-websocket/src/main/java/org/jivesoftware/smack/websocket/XmppWebsocketTransportModule.java

```diff
@@ -0,0 +1,332 @@
+            // Assert that there are no stale discovered endpoints prior performing the lookup.
+            assert discoveredWebsocketEndpoints == null;
+
+            InternalSmackFuture<LookupConnectionEndpointsResult, Exception> websocketEndpointsLookupFuture = new InternalSmackFuture<>();
+
+            connectionInternal.asyncGo(() -> {
+
+                WebsocketRemoteConnectionEndpoint providedEndpoint = null;
+
+                // Check if there is a websocket endpoint already configured.
+                URI uri = moduleDescriptor.getExplicitlyProvidedUri();
+                if (uri != null) {
+                    providedEndpoint = new WebsocketRemoteConnectionEndpoint(uri);
+                }
+
+                if (!moduleDescriptor.isWebsocketEndpointDiscoveryEnabled()) {
+                    // If discovery is disabled, assert that the provided endpoint isn't null.
+                    assert providedEndpoint != null;
+
+                    SecurityMode mode = connectionInternal.connection.getConfiguration().getSecurityMode();
+                    if ((providedEndpoint.isSecureEndpoint() &&
+                            mode.equals(SecurityMode.disabled))
+                            || (!providedEndpoint.isSecureEndpoint() &&
+                                    mode.equals(SecurityMode.required))) {
+                        throw new IllegalStateException("Explicitly configured uri: " + providedEndpoint.getWebsocketEndpoint().toString()
+                                + " does not comply with the configured security mode:" + mode);
+                    }
+
+                    // Generate Result for explicitly configured endpoint.
+                    Result manualResult = new Result(Arrays.asList(providedEndpoint), null);
+
+                    LookupConnectionEndpointsResult endpointsResult = new DiscoveredWebsocketEndpoints(manualResult);
+
+                    websocketEndpointsLookupFuture.setResult(endpointsResult);
+                } else {
+                    DomainBareJid host = connectionInternal.connection.getXMPPServiceDomain();
+                    ModularXmppClientToServerConnectionConfiguration configuration = connectionInternal.connection.getConfiguration();
+                    SecurityMode mode = configuration.getSecurityMode();
+
+                    // Fetch remote endpoints.
+                    Result xep0156result = WebsocketRemoteConnectionEndpointLookup.lookup(host, mode);
+
+                    List<WebsocketRemoteConnectionEndpoint> discoveredEndpoints = xep0156result.discoveredRemoteConnectionEndpoints;
+
+                    int discoveredEndpointCount = xep0156result.discoveredRemoteConnectionEndpoints.size();
+                    List<WebsocketRemoteConnectionEndpoint> endpoints = new ArrayList<>(discoveredEndpointCount);
+
+                    // Add remote discovered endpoints.
+                    endpoints.addAll(discoveredEndpoints);
```

Comment:
I believe you can replace this two lines with 
```
List<WebsocketRemoteConectionEndpoint> endpoints = new ArrayList<>(discoveredEnpoints);
```
and get rid of the size variable.